#!/usr/bin/env python

PACKAGE = "dcsl_vision_tracker"
# import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("binary_threshold", int_t, 0, "The cutoff value for thresholding to a binary image", 75, 0, 255)
gen.add("erode_iterations", int_t, 0, "Level of eroding to get rid of small contours", 0, 0, 10)
gen.add("min_blob_size", int_t, 0, "Minimum area of the robot in square pixels", 200, 0, 3000)
gen.add("max_blob_size", int_t, 0, "Maximum area of the robot in square pixels", 350, 0, 3000)
gen.add("black_threshold", int_t, 0, "Maximum value to consider black", 150, 0, 255)
gen.add("diff_dilations", int_t, 0, "Level of dilation to expand diff holes", 3, 0, 10)
# Could use enumeration technique to do image size

exit(gen.generate(PACKAGE, "dcsl_miabot_tracker.py", "dcsl_miabot_tracker_config"))
